File: Customer.java
public class Customer
{
   static final int SIZE = 150;
   static Customer [] database = new Customer [SIZE];
   static int size = 0;
   private String name;
   private String password;
   private Chequing chequingAccount;
   // Savings savingsAccount, Investment investmentAccount; 

   public Customer(String name, String password)
   {
      this.name = name;
      this.password = password;
      chequingAccount = null;
      database[size++] = this;
   } 

   /****************************
   //
   //
   **************************/
  public String getName()
  {
    return name;
  }

   public String getpassword()
   {
     return password;
   }



   

   /****************************
   // searches for customers information
   // using binary search 
   **************************/
   public boolean search (String password)
   {
      boolean find = false;
      int i = 0;
      while(i<size && !find)
      {
        if((database[i].password).equals(password))
        {
          find = true;
        }
        i++;
      }
      return find;
   }
   ///////////////////////////////////////////////////////////////////////////
   /********************* 
   //Chequing account information
   //Chequing methods are 
   //createChequings()
   //depositChequings()
   //withdrawChequings
   //closeChequings
   //isChequingOpen()
   **********************/
   public void createChequings()
   {
     chequingAccount =  new Chequing(name, password); 
   }
   

   public void depositChequings(int amount)
   {
      chequingAccount.deposit(amount);
      System.out.println("Your new balance is $" + chequingAccount.getCurrentAmount());
   }

   public void withdrawChequings(int amount)
   {
    chequingAccount.withdraw(amount);
    System.out.println("Your new balance is $" + chequingAccount.getCurrentAmount());
   }
   
   public void closeChequings()
   {
     int amount = chequingAccount.getCurrentAmount();
     System.out.println("Your Chequings account has a balance of $" + amount);
     chequingAccount.withdraw(amount);
     chequingAccount = null;
     System.out.println("Your Chequings account has been closed with a $0");
   }

   public boolean isChequingsOpen(Customer cx)
   {
     
     boolean validAccount = false;
     if(cx.chequingAccount != null)
     {
        validAccount = true;
     }
     return validAccount; 
   }

   
}

File: Customer.js
class Customer {
    
    static database = new Array(150);
    static size = 0;
    
    
    constructor(name, password) {
    this.name = name;
    this.password = password;
    this.chequingAccount = null;
    database[size++] = this;
    }
    getName() {
    return this.name;
    }
    
    getPassword() {
    return this.password;
    }
    
    search(password) {
    let find = false;
    let i = 0;
    while (i < size && !find) {
    if (database[i].getPassword() === password) {
    find = true;
    }
    i++;
    }
    return find;
    }
    
    createChequings() {
    this.chequingAccount = new Chequing(this.name, this.password);
    }
    
    depositChequings(amount) {
    this.chequingAccount.deposit(amount);
    console.log('Your new balance is $${this.chequingAccount.getCurrentAmount()}');
    }
    
    withdrawChequings(amount) {
    this.chequingAccount.withdraw(amount);
    console.log('Your new balance is $${this.chequingAccount.getCurrentAmount()}');
    }
    
    closeChequings() {
    let amount = this.chequingAccount.getCurrentAmount();
    console.log('Your Chequings account has a balance of $${amount}');
    this.chequingAccount.withdraw(amount);
    this.chequingAccount = null;
    console.log('Your Chequings account has been closed with a $0');
    }
    
    isChequingsOpen() {
    return this.chequingAccount !== null;
    }
    }
    
    class Chequing extends Customer {
    constructor(name, password) {
    super(name, password);
    this.totalDepositAmount = 0;
    this.currentAmount = 50;
    this.totalAmountTransferred = 0;
    this.totalAmountWithdrawn = 0;
    }
    
    deposit(depositAmount) {
    this.currentAmount += depositAmount;
    this.totalDepositAmount += depositAmount;
    }
    
    withdraw(amountWithdrawn) {
    this.currentAmount -= amountWithdrawn;
    this.totalAmountWithdrawn += amountWithdrawn;
    }
    
    transfer(amount) {
    this.currentAmount -= amount;
    this.totalAmountTransferred += amount;
    }
    
    getCurrentAmount() {
    return this.currentAmount;
    }
    
    getTotalDepositAmount() {
    return this.totalDepositAmount;
    }
    
    getTotalAmountTransferred() {
    return this.totalAmountTransferred;
    }
    
    getTotalAmountWithdrawn() {
    return this.totalAmountWithdrawn;
    }
    

    }
    
    
    
    